whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -156,6 +159,69 @@ public Object mapSearch(final String requestUrl, final List<RpslObject> objects,
         return mapCommons(searchResult, requestUrl);
     }
 
+    public Object mapOrganisationEntity(final String requestUrl, List<RpslObject> objects,
+                                        final Iterable<LocalDateTime> localDateTimes, final int maxResultSize){
+
+
+        List<Autnum> autnums = new ArrayList<>();
+        final Iterator<LocalDateTime> iterator = localDateTimes.iterator();
+        Notice outOfLimitNotice = null;
+
+        if(objects.size() > maxResultSize){
+            objects = getTopLevelObjects(objects);
+
+            outOfLimitNotice = new Notice();
+            outOfLimitNotice.setTitle(String.format("limited search results to %s maximum" , maxResultSize));
+        }
+
+        final List<Ip> ipv4Networks = new ArrayList<>();
+        final List<Ip> ipv6Networks = new ArrayList<>();
+        RdapObject organisation = new RdapObject();
+        for (final RpslObject object : objects) {
+            if (object.getType() == AUT_NUM ) {
```

Comment:
Use `switch(object.getType())` as it's more readable. Also throw IllegalStateException in the default case.